src/Microsoft.OData.Core/ODataContextUrlInfo.cs

```diff
@@ -398,11 +399,12 @@ private static string CreateSelectExpandContextUriSegment(SelectExpandClause sel
         /// <summary>Process sub expand node, contact with subexpand result</summary>
         /// <param name="expandNode">The current expanded node.</param>
         /// <param name="subExpand">Generated sub expand node.</param>
+        /// <param name="version">OData Version of the generated response.</param>
         /// <returns>The generated expand string.</returns>
-        private static string ProcessSubExpand(string expandNode, string subExpand)
+        private static string ProcessSubExpand(string expandNode, string subExpand, ODataVersion version)
         {
-
-            return expandNode + ODataConstants.ContextUriProjectionStart + subExpand + ODataConstants.ContextUriProjectionEnd;
+            return string.IsNullOrEmpty(subExpand) && version <= ODataVersion.V4 ? expandNode :

```

Comment:
>string.IsNullOrEmpty(subExpand) [](start = 19, length = 31)

why add IsNullOrEmpty(subExpand) test?

Reply:
In V4.01, we need to have empty parentheses in the context URI for an expanded navigation property even if there are not nested select or expand option. With Biao's change, we started adding the empty parentheses for V4 as well. The check in question basically determines if there was any nested expand or select for the current expand in the select expand.

---
In reply to: [256179761](https://github.com/OData/odata.net/pull/1402#discussion_r256179761) [](ancestors = 256179761)